login/service.go

```diff
@@ -82,11 +81,16 @@ var allProvidersToLink = []string{"github", "openshift-v3"}
 
 const (
 	initiateLinkingParam = "initlinking"
+	apiTokenParam        = "api_token"
 )
 
-// Perform performs authentication
-func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.LoginLoginContext, config *oauth2.Config, serviceConfig LoginServiceConfiguration) error {
+type OauthConfig interface {
+	Exchange(ctx netcontext.Context, code string) (*oauth2.Token, error)
+	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
```

Comment:
Should we call it  `AuthorizationGrant`  to align it with Oauth terminologies.

Reply:
I called this OuathConfig bc I needed some interface to replace the oauth2.Config implementation by some interface.

Reply:
No, that part's fine. Could we replace `AuthCodeURL` with `AuthorizationGrant` .

Reply:
Oh, no, we can't :) It's how they call this method in oauth2.Config. I had to use our own interface here instead of oauth2.Config because I needed to pass a dummy implementation here in tests.
So, I added two methods to this interface from the original oauth2.Config type. The methods which we use.